src/Mvc/Mvc.Core/src/ModelBinding/Binders/ComplexObjectModelBinder.cs

```diff
@@ -75,32 +75,33 @@ private async Task BindModelCoreAsync(ModelBindingContext bindingContext, int pr
             var bindingSucceeded = false;
 
             var modelMetadata = bindingContext.ModelMetadata;
+            var boundConstructor = modelMetadata.BoundConstructor;
 
-            if (bindingContext.Model == null)
+            if (boundConstructor != null)
             {
-                var boundConstructor = modelMetadata.BoundConstructor;
-                if (boundConstructor != null)
-                {
-                    var values = new object[boundConstructor.BoundConstructorParameters.Count];
-                    var (attemptedParameterBinding, parameterBindingSucceeded) = await BindParametersAsync(
-                        bindingContext,
-                        propertyData,
-                        boundConstructor.BoundConstructorParameters,
-                        values);
+                // To bind record we need to instantiate the type. This means we'll ignore a previously
+                // assigned bindingContext.Model value.
+                // This behaior is identical to input formatting with S.T.Json and Json.NET.
```

Comment:
nit: Suggest mentioning records in this comment. That saves looking at the `BoundConstructor` doc comments.